google/cloud/sqlalchemy_spanner/sqlalchemy_spanner.py

```diff
@@ -38,6 +40,13 @@
 from google.cloud import spanner_dbapi
 from google.cloud.sqlalchemy_spanner._opentelemetry_tracing import trace_call
 
+
+@listens_for(Pool, "reset")
+def reset_connection(dbapi_conn, connection_record):
+    """An event of returning a connection back to a pool."""
+    dbapi_conn.connection.staleness = None
```

Comment:
According to the docs, connection the hook to the `Pool` class will connect it to all the pools used by the dialect. In the test below a `QueuedPool` is used (by default), and it's working fine.